pkgs/by-name/gk/gk-cli/package.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  coreutils,
+  installShellFiles,
+  makeWrapper,
+}:
+
+let
+  version = "2.1.0";
+in
+stdenv.mkDerivation {
+  pname = "gk-cli";
+  inherit version;
+
+  src = fetchzip {
+    url = "https://github.com/gitkraken/gk-cli/releases/download/v${version}/gk_${version}_Linux_x86_64.zip";
+    sha256 = "sha256-BZO8MVFxIXXB+9v4yB9XcZ0E68xOIUL7Mot7PZ+5M1U=";
+    stripRoot = false;
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ];
+
+  postInstall = ''
+    mkdir --parents $out/bin/
+    install ./gk $out/bin/
```

Install can create the destination paths as necessary using the `-D` option:
```suggestion
    install -D ./gk $out/bin/
```

---

I tried using the following command but i got an error

```sh
NIXPKGS_ALLOW_UNFREE=1 nix build --impure -L .#gk-cli
```

<details>

<summary>very long log</summary>

```txt
warning: Git tree '/home/pinage404/Project/nixpkgs' is dirty
gk-cli> sourcing setup hook '/nix/store/lixqzq525cw2l89z3nsayp2rmj1c93rg-install-shell-files/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/mf179bxvndj9svmq6ds2p7yrlkvgh4y2-make-shell-wrapper-hook/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/b08ng3862qcwrhkx8p2ji563pr87har8-die-hook/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/dv5vgsw8naxnkcc88x78vprbnn1pp44y-patchelf-0.15.0/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/i4iynx9axbq23sd0gyrc5wdb46zz6z8l-update-autotools-gnu-config-scripts-hook/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh'
gk-cli> sourcing setup hook '/nix/store/m54bmrhj6fqz8nds5zcj97w9s9bckc9v-compress-man-pages.sh'
gk-cli> sourcing setup hook '/nix/store/wgrbkkaldkrlrni33ccvm3b6vbxzb656-make-symlinks-relative.sh'
gk-cli> sourcing setup hook '/nix/store/5yzw0vhkyszf2d179m0qfkgxmp5wjjx4-move-docs.sh'
gk-cli> sourcing setup hook '/nix/store/fyaryjvghbkpfnsyw97hb3lyb37s1pd6-move-lib64.sh'
gk-cli> sourcing setup hook '/nix/store/kd4xwxjpjxi71jkm6ka0np72if9rm3y0-move-sbin.sh'
gk-cli> sourcing setup hook '/nix/store/pag6l61paj1dc9sv15l7bm5c17xn5kyk-move-systemd-user-units.sh'
gk-cli> sourcing setup hook '/nix/store/jivxp510zxakaaic7qkrb7v1dd2rdbw9-multiple-outputs.sh'
gk-cli> sourcing setup hook '/nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh'
gk-cli> sourcing setup hook '/nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh'
gk-cli> sourcing setup hook '/nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh'
gk-cli> sourcing setup hook '/nix/store/aazf105snicrlvyzzbdj85sx4179rpfp-set-source-date-epoch-to-latest.sh'
gk-cli> sourcing setup hook '/nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh'
gk-cli> sourcing setup hook '/nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0/nix-support/setup-hook'
gk-cli> sourcing setup hook '/nix/store/pg90p34kys2famxnq7925sbgj4jrnsi8-binutils-wrapper-2.42/nix-support/setup-hook'
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/lixqzq525cw2l89z3nsayp2rmj1c93rg-install-shell-files
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/lixqzq525cw2l89z3nsayp2rmj1c93rg-install-shell-files
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/mf179bxvndj9svmq6ds2p7yrlkvgh4y2-make-shell-wrapper-hook
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/mf179bxvndj9svmq6ds2p7yrlkvgh4y2-make-shell-wrapper-hook
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/b08ng3862qcwrhkx8p2ji563pr87har8-die-hook
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/b08ng3862qcwrhkx8p2ji563pr87har8-die-hook
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/dv5vgsw8naxnkcc88x78vprbnn1pp44y-patchelf-0.15.0
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/dv5vgsw8naxnkcc88x78vprbnn1pp44y-patchelf-0.15.0
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/i4iynx9axbq23sd0gyrc5wdb46zz6z8l-update-autotools-gnu-config-scripts-hook
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/i4iynx9axbq23sd0gyrc5wdb46zz6z8l-update-autotools-gnu-config-scripts-hook
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/m54bmrhj6fqz8nds5zcj97w9s9bckc9v-compress-man-pages.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/m54bmrhj6fqz8nds5zcj97w9s9bckc9v-compress-man-pages.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/wgrbkkaldkrlrni33ccvm3b6vbxzb656-make-symlinks-relative.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/wgrbkkaldkrlrni33ccvm3b6vbxzb656-make-symlinks-relative.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/5yzw0vhkyszf2d179m0qfkgxmp5wjjx4-move-docs.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/5yzw0vhkyszf2d179m0qfkgxmp5wjjx4-move-docs.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/fyaryjvghbkpfnsyw97hb3lyb37s1pd6-move-lib64.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/fyaryjvghbkpfnsyw97hb3lyb37s1pd6-move-lib64.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/kd4xwxjpjxi71jkm6ka0np72if9rm3y0-move-sbin.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/kd4xwxjpjxi71jkm6ka0np72if9rm3y0-move-sbin.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/pag6l61paj1dc9sv15l7bm5c17xn5kyk-move-systemd-user-units.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/pag6l61paj1dc9sv15l7bm5c17xn5kyk-move-systemd-user-units.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/jivxp510zxakaaic7qkrb7v1dd2rdbw9-multiple-outputs.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/jivxp510zxakaaic7qkrb7v1dd2rdbw9-multiple-outputs.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/aazf105snicrlvyzzbdj85sx4179rpfp-set-source-date-epoch-to-latest.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/aazf105snicrlvyzzbdj85sx4179rpfp-set-source-date-epoch-to-latest.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0
gk-cli> calling 'envHostHostHook' function hook 'ccWrapper_addCVars' /nix/store/pg90p34kys2famxnq7925sbgj4jrnsi8-binutils-wrapper-2.42
gk-cli> calling 'envHostHostHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/pg90p34kys2famxnq7925sbgj4jrnsi8-binutils-wrapper-2.42
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/lixqzq525cw2l89z3nsayp2rmj1c93rg-install-shell-files
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/lixqzq525cw2l89z3nsayp2rmj1c93rg-install-shell-files
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/mf179bxvndj9svmq6ds2p7yrlkvgh4y2-make-shell-wrapper-hook
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/mf179bxvndj9svmq6ds2p7yrlkvgh4y2-make-shell-wrapper-hook
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/b08ng3862qcwrhkx8p2ji563pr87har8-die-hook
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/b08ng3862qcwrhkx8p2ji563pr87har8-die-hook
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/dv5vgsw8naxnkcc88x78vprbnn1pp44y-patchelf-0.15.0
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/dv5vgsw8naxnkcc88x78vprbnn1pp44y-patchelf-0.15.0
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/i4iynx9axbq23sd0gyrc5wdb46zz6z8l-update-autotools-gnu-config-scripts-hook
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/i4iynx9axbq23sd0gyrc5wdb46zz6z8l-update-autotools-gnu-config-scripts-hook
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/m54bmrhj6fqz8nds5zcj97w9s9bckc9v-compress-man-pages.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/m54bmrhj6fqz8nds5zcj97w9s9bckc9v-compress-man-pages.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/wgrbkkaldkrlrni33ccvm3b6vbxzb656-make-symlinks-relative.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/wgrbkkaldkrlrni33ccvm3b6vbxzb656-make-symlinks-relative.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/5yzw0vhkyszf2d179m0qfkgxmp5wjjx4-move-docs.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/5yzw0vhkyszf2d179m0qfkgxmp5wjjx4-move-docs.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/fyaryjvghbkpfnsyw97hb3lyb37s1pd6-move-lib64.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/fyaryjvghbkpfnsyw97hb3lyb37s1pd6-move-lib64.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/kd4xwxjpjxi71jkm6ka0np72if9rm3y0-move-sbin.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/kd4xwxjpjxi71jkm6ka0np72if9rm3y0-move-sbin.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/pag6l61paj1dc9sv15l7bm5c17xn5kyk-move-systemd-user-units.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/pag6l61paj1dc9sv15l7bm5c17xn5kyk-move-systemd-user-units.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/jivxp510zxakaaic7qkrb7v1dd2rdbw9-multiple-outputs.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/jivxp510zxakaaic7qkrb7v1dd2rdbw9-multiple-outputs.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/aazf105snicrlvyzzbdj85sx4179rpfp-set-source-date-epoch-to-latest.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/aazf105snicrlvyzzbdj85sx4179rpfp-set-source-date-epoch-to-latest.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/62zpnw69ylcfhcpy1di8152zlzmbls91-gcc-wrapper-13.3.0
gk-cli> calling 'envHostTargetHook' function hook 'ccWrapper_addCVars' /nix/store/pg90p34kys2famxnq7925sbgj4jrnsi8-binutils-wrapper-2.42
gk-cli> calling 'envHostTargetHook' function hook 'bintoolsWrapper_addLDVars' /nix/store/pg90p34kys2famxnq7925sbgj4jrnsi8-binutils-wrapper-2.42
gk-cli> Running phase: unpackPhase
gk-cli> unpacking source archive /nix/store/na8pa2p6f3xw1lk8282ybwq9jvx3kln0-source
gk-cli> calling 'unpackCmd' function hook '_defaultUnpack' /nix/store/na8pa2p6f3xw1lk8282ybwq9jvx3kln0-source
gk-cli> source root is source
gk-cli> calling 'postUnpack' function hook '_updateSourceDateEpochFromSourceRoot'
gk-cli> Running phase: patchPhase
gk-cli> Running phase: updateAutotoolsGnuConfigScriptsPhase
gk-cli> Running phase: configurePhase
gk-cli> calling 'preConfigure' function hook '_multioutConfig'
gk-cli> no configure script, doing nothing
gk-cli> Running phase: buildPhase
gk-cli> no Makefile or custom buildPhase, doing nothing
gk-cli> Running phase: installPhase
gk-cli> install: cannot create regular file '/nix/store/phccpam6wsj230pccdswxds1bzj5vbq4-gk-cli-2.1.0/bin/': Not a directory
```

</details>

I don't understand what i did wrong

---

Apologies my suggestion was incomplete; the `install` commands also the `-t` option for `-D` to work properly it seems.

Here's how I'd write the `installPhase` for this package:

```nix
  installPhase = ''
    runHook preInstall

    install -Dm555 gk -t $out/bin/

    installShellCompletion --bash ./**/gk.bash
    installShellCompletion --fish ./**/gk.fish
    installShellCompletion --zsh ./**/_gk

    runHook postInstall
  '';
```